pkg/proc/stack.go

```diff
@@ -542,6 +425,7 @@ func (it *stackIterator) advanceRegs() (callFrameRegs op.DwarfRegisters, ret uin
 	} else {
 		framectx = it.bi.Arch.FixFrameUnwindContext(fde.EstablishFrame(it.pc), it.pc, it.bi)
 	}
+	it.bi.Arch.SwitchSp(it)
```

Comment:
Why do we only want to switch stack pointer here? Is there a reason to special case this and not just use `SwitchStack`?

Reply:
The `SwitchSp` is a part of `advanceRegs`, and it must get right SP value before get other regs. But the `SwitchStack` is called after all regs got in `Next`. That's the reason we need `SwitchSp` there.